aimon/decorators/detect.py

```diff
@@ -31,18 +33,27 @@ class DetectResult:
         Returns a string representation of the DetectResult object (same as __str__).
     """
 
-    def __init__(self, status, detect_response, publish=None):
+    def __init__(self, status, detect_response, publish=None, async_mode=None):
         self.status = status
-        self.detect_response = detect_response
+        self.async_mode = async_mode
         self.publish_response = publish if publish is not None else []
 
+        # Format or convert detect_response during initialization
+        if self.async_mode:
+            # Convert detect_response dictionary to an object with dot notation
+            self.detect_response = type('Obj', (object,), detect_response)()
+        else:
+            # Format detect_response for string representation
+            detect_response_value = (
+                detect_response.to_dict() if hasattr(detect_response, 'to_dict') else detect_response
+            )
+            self.detect_response = self._format_response_item(detect_response_value)
+
     def __str__(self):
-        # Use format_response_item to format detect_response
-        detect_response_str = self._format_response_item(self.detect_response.to_dict())
         return (
             f"DetectResult(\n"
             f"  status={self.status},\n"
-            f"  detect_response={detect_response_str},\n"
+            f"  detect_response={self.detect_response},\n"
```

Comment:
instead format the object in a pretty string format here using `_format_response_item`